src/compiler/scala/tools/nsc/transform/ErasureApi.scala

```diff
@@ -0,0 +1,26 @@
+package scala.tools.nsc
+package transform
+
+trait Erasure extends typechecker.Typers 
+  with scala.reflect.internal.transform.Erasure
+  with typechecker.Contexts
+  with typechecker.Globals { self:
+  // required by typechecker.Typers
+  //typechecker.Globals with
+  //typechecker.Contexts
+  typechecker.ContextErrors with
+  typechecker.Infer with
+  // required by typechecker.Contexts
+  //typechecker.ContextErrors with
+  typechecker.Implicits with
+  typechecker.NamesDefaults =>
+
+  import global._
+
+  private [nsc] def minimizeParents(parents: List[Type]): List[Type]
+  private [nsc] def javaSig(sym0: Symbol, info: Type): Option[String]
```

Comment:
This method is called from the IDE.

```
% git grep erasure.javaSig
org.scala-ide.sdt.core/src/org/scalaide/core/internal/compiler/JavaSig.scala:          for (signature <- erasure.javaSig(symbol, pc.enteringPhase(pc.currentRun.erasurePhase)(symbol.info)))
```

In general, be careful chaning `public` to `private[nsc]` for this reason.

Once you're ready, we can your branch through our PR validation and community build that will check that the compiler clients like Scala IDE, SBT, Ensime and Macro Paradise still work.